### ISSUE TITLE
References to input elements behind form components

### ISSUE THREAD

**Issue by Pezmc** [flowforge/forge-ui-components#60]
### As a:

Developer

### I want to:

Have a reference to the underlying input DOM element 

### So that:

It can be manipulated directly, for example, to call `.focus()` or read data using a `FileReader`.

----

Right now there's no way to reference the underlying input elements for any of the form components. 

Pre `forge-ui-components` it looks like an ID would be set as part of `FormRow` that was then passed down, and used to access the elements.

https://github.com/flowforge/flowforge/blob/0dc8b449036ed3310b8d1c3ab9337a057ca174c6/frontend/src/components/FormRow.vue#L85-L87

https://github.com/flowforge/flowforge/blob/0dc8b449036ed3310b8d1c3ab9337a057ca174c6/frontend/src/pages/project/Settings/General.vue#L76-L78

Being able to reference the underlying DOM element would allow this to be implemented in a Vue.js'y way (via `this.$refs.FormRow.$refs.input` for example).

Alternatively:

- an autofocus attribute could be implemented that calls focus (and/or uses the autofocus element attribute), however, this won't cover all use cases, e.g. username re-focus when login fails.

https://github.com/flowforge/flowforge/blob/0dc8b449036ed3310b8d1c3ab9337a057ca174c6/frontend/src/components/auth/Credentials.vue#L72-L75

- A `focus()` method could be added to all form components in forge-ui-components (https://github.com/flowforge/forge-ui-components/issues/42)

Related to https://github.com/flowforge/flowforge/issues/1081

### Acceptance Criteria

- [ ] Consumers of forge-ui-components can access input elements
- [ ] It's possible to call `.focus()` on a forge-ui-component form element

**Comment by joepavitt** [comment]
There is value in having an `autofocus` property still though, so we could scope this issue to that for now.

**Comment by Pezmc** [comment]
@joepavitt and I have bounced back and forth a bit on slack.

I was hesitant that we'll end up having to re-implement _all_ input element methods (e.g. blur/click/focus/select etc) on each of the form components if we start with `.focus()` (#42).

However, @joepavitt makes the point that we want both `forge-ui-components` elements (and FlowForge/FormRow) to be as easy as possible to use. We also looked at a few other Vue.js component libraries, and it seems like there is precedent to re-implement methods like `focus`/`blur`/`select`.

To that end, we're going to start with implementing `focus()` rather than adding references and revisit if it feels we have too many methods or need references for another reason. I'm closing out the issue and opening another for the autofocus property.
